Avaritia Renders for MetaItems and Materials

jude123412 opened the pull request:
## What
I have added some Avaritia Methods to GT MetaItems and GT Materials.
This allows for all Material items and MetaItems to use Avaritia render's like Halo's behind items and Cosmic Effect for items.
Making Materials like Infinity or Cosmic Neutronium are missing these methods inside the Material Builder To give them Halo noise, Pulse effect, Halo, Halo Colour, Item Pulsing or Cosmic Effect. These new Renders are only active when Avaritia is installed because I'm using Avaritia to render everything.

I think what I have listed above is a good enough reason to accept this PR, This is somthing that I desire and many other do too.

## Implementation Details
The MetaItem, Material MaterialBuilder should be checked they are where most of the methods are but other things in this PR could be checked too.

## Outcome
This Pr will add support for Avaritia Renders Such as shouldDrawHalo, haloTexture, haloColour, haloSize, shouldDrawPulse, shouldDrawCosmic/maskTexturePath and maskOpacity to MetaItems and Materials in the Material Builder.

These Methods will only become active when Avaritia Is installed, When it is not the Methods do nothing.

## Additional Information
These are a couple of the new render methods for Materials and MetaItems I made using the methods stated above.

Avaritia Halo and Cosmic Effect for NotANoob Certificate

![image](https://github.com/user-attachments/assets/59094b11-daf6-428d-bc03-bd14da280b04)

Test Material With green coloured halo and Pulse Effect

https://github.com/user-attachments/assets/54d741a5-b169-47ac-b2ef-53f2b3f4dde1

Universium With Cosmic Effect

https://github.com/user-attachments/assets/dcb1761d-aeb1-4134-b0fd-9b1991ed95b1

These are the Methods you can define inside the Material Builder.
They are all optional methods that are only active when Avaritia is installed.
.cosmic(boolean shouldDrawHalo, String haloTexture, String haloColour, int haloSize, boolean shouldDrawPulse, boolean shouldDrawCosmic, Float maskOpacity) 
cosmic(boolean shouldDrawHalo, String haloTexture, String haloColour, int haloSize, boolean shouldDrawPulse)
cosmic(boolean shouldDrawHalo, String haloTexture, String haloColour, int haloSize) 
cosmic( boolean shouldDrawCosmic, Float maskOpacity) 
cosmic(boolean shouldDrawPulse)

shouldDrawHalo will set a Material set to draw a Halo, This will need also need haloTexture, haloColour and haloSize to be set.

haloTexture will get the halo from "gregtech:texture/items/cosmic/String haloTexture".

haloColour What colour the Halo will be as a Hex Value "FF00FF00" for Max Alpha and green

haloSize The size of the halo Max value of 10. defined as an Integer, 10 would be max size.

shouldDrawCosmic will generate the correct locations for textures of the Material Set, If the IconSet is set to DULL for instance, all Mask Textures are generated under "gregtech:textures/items/material_sets/dull/%s_mask." (%s is the item type, dust, ingot, gem).

maskOpacity Defines the opacity of the mask as a float, Max value 1.0f

These are the Methods you can define for any MetaItem.
They are all optional methods that are only active when Avaritia is installed.
.cosmicProperties(boolean shouldDrawHalo, String haloTexture, String haloColour, int haloSize, boolean shouldDrawPulse, String maskTexture, float maskOpacity)
.cosmicProperties(boolean shouldDrawHalo, String haloTexture, String haloColour, int haloSize, boolean shouldDrawPulse)
.cosmicProperties(boolean shouldDrawHalo, String haloTexture, String haloColour, int haloSize)
.cosmicProperties(String maskTexture, Float maskOpacity)
.cosmicProperties(boolean shouldDrawPulse)

These work the same as the Material builder accept they are on a per item basis and the mask Texture is defined as a string and will be under "gregtech:texture/items/cosmic/mask/String maskTexture".

## Potential Compatibility Issues
Avaritia 1.1x wont work with these renders because that version doesn't have the ability to create an item with Cosmic and Halo Effects. Avaritia 1.1x Unofficial Extended Life contains everthing required for this to work as long as the version is greater than 3.3.0.42. Don't worry, If you have Avaritia 1.1x and not Avaritia 1.1x UEL these methods do nothing and won't crash the game.

serenibyss committed:
Add IControllable to creative energy emitter (#2248)

TechLord22 committed:
fix using client only method for harder bed recipe names (#2253)

ALongStringOfNumbers committed:
Fix plasma temps all being the default temp (#2256)

serenibyss committed:
Update build script version to 1701739812 (#2257)

Co-authored-by: DStrand1 <DStrand1@users.noreply.github.com>

Tictim committed:
Re-implement walking speed bonus (#2251)

ghzdude committed:
Rework Generators 2.0 ™️  (#2255)

serenibyss committed:
Fix issues with ore byproduct page (#2262)

serenibyss committed:
Clean up and add particle effects for machines (#2249)

serenibyss committed:
Fix fluid output trim losing some fluids (#2268)

TechLord22 committed:
fix crash with fluid filters with null delegate containers (#2273)

TechLord22 committed:
fix data stick entry removal from AL (#2279)

ghzdude committed:
Fix Large Gas Turbine not displaying Max Voltage when fully spun up (#2283)

serenibyss committed:
Update build script version to 1702244235 (#2282)

Co-authored-by: DStrand1 <DStrand1@users.noreply.github.com>
Co-authored-by: alongstringofnumbers <sjensen793@gmail.com>

serenibyss committed:
2.8.3

serenibyss committed:
Fix addFluid expansion method for GrS/CT (#2287)

serenibyss committed:
More tool compat (#2292)

serenibyss committed:
Fix silicon bee crash (#2299)

serenibyss committed:
Fix using MV muffler in EBF JEI preview (#2298)

ALongStringOfNumbers committed:
Allow specifying NBT matching on Research ItemStacks (#2294)

ALongStringOfNumbers committed:
Attempt to clarify tooltip (#2300)

serenibyss committed:
Chisel module (#2290)

serenibyss committed:
Fix pipe restrictor render on bottom face (#2302)

serenibyss committed:
Fix battery slot charge flickering (#2305)

serenibyss committed:
Tweak muffler effect API (#2286)

TechLord22 committed:
log incomplete packet reads in NetworkHandler (#2288)

serenibyss committed:
Fusion rebalancing (#2252)

serenibyss committed:
Fix plasma temp NPE for plasma-only materials (#2306)

Tictim committed:
Ore Dictionary Filter Update (#2296)

TechLord22 committed:
invalidate pipenet routepaths on chunk unload (#2303)

serenibyss committed:
Allow electrolyzing more hydrocarbons (#2258)

serenibyss committed:
2.8.4

TechLord22 committed:
only access worldpipenet on server (#2310)

serenibyss committed:
2.8.5

TechLord22 committed:
update GHA workflows (#2270)

ALongStringOfNumbers committed:
Fix a couple incomplete recieveCustomData calls (#2322)

TechLord22 committed:
fix test and formatting action concurrency groups

serenibyss committed:
Add some GT items to ExtraBees alveary mutator (#2315)

serenibyss committed:
Update build script version to 1702805890 (#2309)

Co-authored-by: DStrand1 <DStrand1@users.noreply.github.com>
Co-authored-by: alongstringofnumbers <sjensen793@gmail.com>

josiah-roberts committed:
Fix large turbine recipe selection (#2324)

josiah-roberts committed:
Use locale formatting where possible in WAILA and The One Probe (#2327)

josiah-roberts committed:
Use EU/t for digital interface (#2329)

marisathewitch committed:
Update ru_ru.lang (#2307)

ALongStringOfNumbers committed:
Improve Lighter Opening (#2323)

TechLord22 committed:
Fix some JEI plugin issues (#2337)

serenibyss committed:
Fix treated wood stick saw recipe (#2339)

iouter committed:
update zh_cn.lang (#2334)

serenibyss committed:
Update build script version to 1704659416 (#2341)

Co-authored-by: DStrand1 <DStrand1@users.noreply.github.com>
Co-authored-by: alongstringofnumbers <sjensen793@gmail.com>

TechLord22 committed:
prevent potential ThreadLocal memory leaks (#2320)

Tictim committed:
Make translation key of MetaItems modifiable (#2259)

Co-authored-by: ALongStringOfNumbers <31759736+ALongStringOfNumbers@users.noreply.github.com>

serenibyss committed:
Update build script version to 1705076830 (#2344)

Co-authored-by: DStrand1 <DStrand1@users.noreply.github.com>

bruberu committed:
Overload AbstractRecipeLogic functions (#2311)

Co-authored-by: alongstringofnumbers <sjensen793@gmail.com>

bruberu committed:
Fix erroring material handlers when liquids don't exist (#2343)

serenibyss committed:
Fix reservoir hatch issues (#2349)

serenibyss committed:
Fix LCE/ECE obstruction check (#2346)

M-W-K committed:
Fix electric tool damage config innacuracy (#2350)

TechLord22 committed:
weaken recipe logic type in WorkableTieredMetaTileEntity (#2342)

ghzdude committed:
Fix Issues with Fusion Reactor Overclocking (#2352)

serenibyss committed:
Separate PSS average IO into in and out (#2353)

TechLord22 committed:
fix gt tile entities not rendering in AE2 pattern outputs (#2319)

serenibyss committed:
Cherrypick of #2325

serenibyss committed:
2.8.6

ghzdude committed:
Fix Player Unable to Climb Frames on Certain Pipes (#2359)

loxoDev committed:
Fix recipe properties offsetting info text too much (#2360)

serenibyss committed:
Fix central monitor jei page (#2364)

ALongStringOfNumbers committed:
Fix unlocalized name of GT fluids in drums (#2361)

serenibyss committed:
Add OpenComputers functions for maintenance problems (#2356)

serenibyss committed:
Add Powderbarrel and ITNT (#2314)

serenibyss committed:
Refactor other mod id handling (#2354)

iouter committed:
Update zh_cn.lang (#2362)

serenibyss committed:
Clean up MetaTileEntities class formatting (#2371)

ghzdude committed:
Set Hardness and Resistance to LDP Blocks (#2370)

serenibyss committed:
Fix hasEnoughPower() check in ARL (#2365)

serenibyss committed:
Use drain/extract for Recipe#matches (#2358)

serenibyss committed:
Fix part sharing tooltip missing on some parts (#2375)

htmlcsjs committed:
re-enable sharing for laser hatches (#2376)

serenibyss committed:
Config and API for tick acceleration blocking (#2381)

TechLord22 committed:
fix material tree crash (#2382)

TechLord22 committed:
fix #2382 (#2386)

serenibyss committed:
Update build script version to 1707682661 (#2380)

serenibyss committed:
Use GTCEu maven (#2379)

serenibyss committed:
Fix parallel overclocks using the wrong voltage for multiblocks (#2390)

serenibyss committed:
Temporary hotfix for research ID lookup (#2391)

TechLord22 committed:
Use chestWood oredict for ULV and LV item buses (#2383)

serenibyss committed:
add maven to GHA publish action

serenibyss committed:
cherrypick fix

serenibyss committed:
2.8.7

serenibyss committed:
fix maven upload action

TechLord22 committed:
fix quantum tank not always fully reading custom data (#2396)

(cherry picked from commit 675efe0712af3a4dcdae53e3075a6dd45fc0c9c9)

M-W-K committed:
Fix NPE from facing being null when drawing grid overlays (#2401)

(cherry picked from commit ae5ef986446366ccb7305bc946e5a01a7affaa92)

YoungOnionMC committed:
Drain Tricorder energy only when in survival (#2402)

(cherry picked from commit 74272b220806d2268f3d5d33443a0e8340ef9d3d)

serenibyss committed:
Advanced Stocking Input Bus + Advanced Stocking Input Hatch (#2367)

(cherry picked from commit db174d2d802cbf7b51bfb9553fc7ec4921fec569)

serenibyss committed:
Electric Wirecutter (#2404)

(cherry picked from commit 4e47ef8e75cab83ffa25dc90b80adf37add2cafd)

serenibyss committed:
Allow PSS to display on CM (#2405)

(cherry picked from commit fd65290173b5de45c22ad1e0398bfd3714b70689)

bruberu committed:
Fix server crashes with alarms (#2411)

(cherry picked from commit 77ed2e04876c133c8d09ac42ef8a46922c9bd6a2)

TechLord22 committed:
fix pipe capability retrieval logic (#2415)

(cherry picked from commit e85169eb34a849d1b56514d4b090cda9080b5bfe)

TechLord22 committed:
fix solar panels not outputting to cables (#2416)

(cherry picked from commit fea6440e9852c1e44a88332b568981d2d19c3b7b)

brachy84 committed:
Update GroovyScript & improve material API for groovy (#2389)

(cherry picked from commit dc389f8f66475dd741ac96138f926e7f94cb0f70)

TechLord22 committed:
allow SMD diodes in low tier diode block recipes (#2423)

(cherry picked from commit c2f681f69f7b61c76e4b113263474133993ad056)

TechLord22 committed:
check for a valid MTE before rendering (#2424)

(cherry picked from commit ba69a484403b01e3d87703546c582b61bfe0d596)

kumquat-ir committed:
Fix link to GT IMPACT in readme (#2427)

(cherry picked from commit 20e61ca2cac5bde02d56827e7a019aa21cb69f5a)

M-W-K committed:
Fix fluid drilling rig display units (#2417)

(cherry picked from commit ad8275f17f3f735867f65ed40ec5c09b76d4309b)

iouter committed:
Update zh_cn.lang (#2413)

(cherry picked from commit db02d99d491e35d00740e338e1feda5046ca7458)

serenibyss committed:
Allow Data Stick to copy-paste ME bus settings (#2419)

(cherry picked from commit 111326a1222de0cd03b18f1cd8a73163110f88cc)

ghzdude committed:
Fix LCE/ECE from parallelizing too much (#2426)

(cherry picked from commit aff043cfaa43f8dd44c45fb5de602fa2d3b56282)

TechLord22 committed:
add ItemStack inputNBT method (#2431)

(cherry picked from commit e6cc4930ca204be09d0721198b274f344c0add4e)

TechLord22 committed:
fix MTE block opacity (#2440)

(cherry picked from commit 323d93a91c390fd12d11bf2971acb6255fcca120)

TechLord22 committed:
minor allocation optimization for GTUtility#canSeeSunClearly (#2441)

(cherry picked from commit 1e98892e6ec28b81d5930fa6a9c0bc62b2ff41cf)

ghzdude committed:
Fix potential MetaTileEntity NPE when calling a server side method on client (#2437)

(cherry picked from commit 1e4ca34ea0b1ed2fc38f9a05c05b5f97d0bd4f5c)

Commit:
Add stairsRecipeName (#2435)

(cherry picked from commit a27b30bb375376c88b62625094cd1e6c43d9eb38)

ghzdude committed:
Improve the Quantum Storage Renderer (#2318)

(cherry picked from commit 06938f05be5bccf67e54e0bb5c9d7b7384c4c75e)

ALongStringOfNumbers committed:
Allow ITNT and PowderBarrels to make AE singularities (#2442)

(cherry picked from commit 7e01d339fb2370eef00ad86b252a3dd00473a338)

TechLord22 committed:
adjust recipe input expected cache size dynamically (#2331)

(cherry picked from commit 251a9ac7306f76205d3505c8febea18066ce5a15)

brachy84 committed:
GroovyScript 1.0.0 (#2425)

Co-authored-by: TechLord22 <37029404+TechLord22@users.noreply.github.com>

(cherry picked from commit 16ed04a0d17f0341d85227e9e9f214ba9e02088f)

TechLord22 committed:
fix primitive multiblocks allowing automation through their controller (#2452)

(cherry picked from commit a8a0ddaa157c78ccba1d544d8789c7d0966b09a9)

TechLord22 committed:
Waila/Hwyla Provider Fixes (#2451)

(cherry picked from commit e927e2d9d54377c96ad2533a9d10e19679abc672)

bruberu committed:
Fix blocks not being able to choose preferred meta values in previews (#2446)

(cherry picked from commit 40b2002c0a20e917be8e0fc4870e65670d064749)

kumquat-ir committed:
Fix display of sifter byproducts for materials with no exquisite (#2444)

(cherry picked from commit e9b942561e3b4aeee4cf0850352928cd382308c7)

vrejhead committed:
Fix possible NPE from MultiblockDisplayText (#2453)

(cherry picked from commit 596a08b521d0a6a4fa3009665b065df434fefb81)

TechLord22 committed:
fix NPE with null fluids in quantum tanks (#2443)

(cherry picked from commit 00032967afeaf39dab06d0a04efc4348b17c731b)

Commit:
Update ja_JP lang for 2.8.8 (#2459)

kumquat-ir committed:
Remove NuclearCraft ASM (#2467)

(cherry picked from commit 43b75c589f84388e70c66a59a879cecfa769be7c)

iouter committed:
Update zh_cn.lang (#2445)

(cherry picked from commit f189493e0216ba80a20a34c23b08d3c26d8ccf7c)

Tictim committed:
Fusion bloom fix & other bloom effect API modifications (#2470)

(cherry picked from commit d006d607f87d000cf444d550ebec217e95911e7c)

htmlcsjs committed:
Inital Vintagium Support (#2463)

Co-authored-by: kumquat-ir <66188216+kumquat-ir@users.noreply.github.com>
(cherry picked from commit 8c2f0c6e72bb5e500552f2ec11309f763275471b)

IntegerLimit committed:
Fix Ore Dict Ingredients When Changing OreDicts with CT & GS (#2456)

(cherry picked from commit e0a246503cd14f012f0f19c4899e3cc4a256de4d)

ALongStringOfNumbers committed:
2.8.8

kumquat-ir committed:
spotless issue

M-W-K committed:
Fix the large turbine rotor texture (#2485)

(cherry picked from commit f8c1a5281e6026a9df22fed3fef644b647e2aae2)

Rundas01 committed:
Small QoL Bee Update (#2458)

(cherry picked from commit 00f3b4dc05800a81e77ddb58b4ec5f01090ccdb7)

M-W-K committed:
Fix excessive packets & re-renders with AEHostable parts (#2487)

(cherry picked from commit 9ce8a202bcad10fa938adb44bfdb4eb632e67b9a)

M-W-K committed:
Allow dynamic hatch placement on Distillation Tower (#2477)

(cherry picked from commit 5c501fd838b1016e988fc48d4ad98decf901f0b4)

Zorbatron committed:
Add a faster recipe for drilling fluid with distilled water (#2484)

(cherry picked from commit 4d3a39efc64552bc992aa939484cf7c149f56867)

TechLord22 committed:
fix pyrolyse speed (#2449)

(cherry picked from commit 09e44d9a4390eff3d2742d0eada0b08db57e9df6)

MCTian-mi committed:
Brute-force Fix of RAM Usage Problem When Using Alfheim (#2475)

(cherry picked from commit ec756bbd72028945780601ffc04266c9135acb36)

M-W-K committed:
Clean up AEHostable part fix (#2488)

(cherry picked from commit 97dd250a10531a1c596c9970bb84475650b9fd0d)

Rundas01 committed:
Updating bees using the new Mutation Condition (#2486)

(cherry picked from commit 72e2ef2aa8ee33004c0fefdeac213942221831c6)

serenibyss committed:
dependency updates

(cherry picked from commit a68d66f38ed40bc3dbe3527663566fa0c8b83421)

serenibyss committed:
Allow recipe logic to set EU/t and speed discounts (#2496)

(cherry picked from commit 132a449a0cc8154be3b79826a4afd6239ead5360)

Zorbatron committed:
Glass tube recycling (#2509)

(cherry picked from commit b22bd7838acda9842ea8e58d3e3cc4b6b52e619e)

M-W-K committed:
Improve assembly line laser render (#2495)

(cherry picked from commit 2811c8ec40496da88bf7648d0f7f3162aa632047)

IntegerLimit committed:
Add Polybenzimidazole Block Conversion Recipes (#2511)

(cherry picked from commit 981fbaa458d303a6f5486e5151771d3308e220d2)

brachy84 committed:
GroovyScript 1.1.0  (#2506)

(cherry picked from commit fe65538edbcb5ad4d7283c82ab1264f819d4c54b)

serenibyss committed:
Update build script version to 1720106721 (#2503)

Co-authored-by: ALongStringOfNumbers <ALongStringOfNumbers@users.noreply.github.com>
(cherry picked from commit e4a86bed2118dd3ce2e82151168fc4a6891e320a)

bruberu committed:
Fix liquid and gas registry key functions overlapping (#2514)

Co-authored-by: TechLord22 <37029404+techlord22@users.noreply.github.com>
(cherry picked from commit 375af9e03c5e8108480a110d9522d538bfbc3938)

TechLord22 committed:
fix failing builds due to Vintagium (#2521)

ALongStringOfNumbers committed:
2.8.9

M-W-K committed:
Improvement to DT hatch logic (#2524)

(cherry picked from commit 3d944344f8b11f82e159645779aa9826be734ad3)

marisathewitch committed:
Update ru_RU lang for 2.8.9 (#2523)

Co-authored-by: Weblate <noreply@weblate.org>
Co-authored-by: Western01 <alexgruzdev47@gmail.com>
(cherry picked from commit a706c39e5213ee557fa67c477e27b766606d326c)

ALongStringOfNumbers committed:
Update Gradle and related files (#2522)

Co-authored-by: TechLord22 <37029404+TechLord22@users.noreply.github.com>
(cherry picked from commit 52c835ce4177ceb8d78755eca43dca5fb12b09b7)

MCTian-mi committed:
Fix the lighter behaviour (#2526)

(cherry picked from commit ae88e46b8387864e27b68401e89f1b767f3372a2)

Synthitic committed:
Fix EUt usage of HPCA with custom higher tier Components (#2512)

(cherry picked from commit 4c1f89fc921cdbb81545166c9ba2abfd3291944c)

TechLord22 committed:
bump GRS and CT versions in gradle (#2528)

(cherry picked from commit f149a18d40464d1bb9d82d385aa4efd336f8323a)

ghzdude committed:
Fix Miner Pipe Deleting TileEntities (#2529)

(cherry picked from commit dd6f3e3a247548daa6d6615683744f79feb4431a)

TechLord22 committed:
Make drums no longer require a Material (#2527)

(cherry picked from commit 69d4e4a793bab4c65ce45a25f9f0b5a1563a9592)

MrKono committed:
Update and Fix ja_jp.lang for 2.8.10 (#2534)

TechLord22 committed:
fix crash with Forestry integration (#2531)

(cherry picked from commit 7799009f8b47bd1bdefd1f11fe3b8c81bf15ef7e)

TechLord22 committed:
2.8.10

jude123412 committed:
Added Avaritia Cosmic Render for MetaItems

jude123412 committed:
MetaPrefixItem Cosmic Effects

Added Cosmic Effect to MetaPrefixItem Class. This allows for all Materials in GregTech to use some Renderers from Avaritia.
Changes to MetaItem and MetaPrefixItem to only allow cosmic effects if Avaritia is installed
Lots of Mask textures for all Material sets

jude123412 committed:
Merge branch 'GTHH-GregTech-2.8.10' of https://github.com/jude123412/GregTech into GTHH-GregTech-2.8.10

jude123412 committed:
Fix Dependencies Post Merge

jude123412 committed:
Update MetaItem.java

Oganesson897 commented:
looks very similar to the code [here](https://gitlab.com/gtcnc/epimorphism-core/-/blob/overhaul/src/main/java/cn/gtcommunity/epimorphism/client/renderer/handler/CosmicItemRenderer.java?ref_type=heads), and with only minor changes, but I dont see any mention of the original author

jude123412 commented:
I think You will need to credit them on this github page or at least Avaritia for using there methods xD

jude123412 committed:
Update CosmicItemRenderer.java

Cleaner Code + Credits to Avaritia Owners

jude123412 committed:
Added Api Files from Avaritia

added Api Files from Avaritia to fix an issue caused by them not being there

jude123412 committed:
SpotlessCheck and SpotlessApply

Run ./gradlew :spotlessCheck --warning-mode all --build-cache and ./gradlew :spotlessApply --warning-mode all to fix and clean code

jude123412 committed:
Update CosmicItemRenderer.java

jude123412 committed:
Create permission5.png

I have received permission from Morpheus1101 to use some of Avaritia's code for this new patch.
Permission5 is what they said.

jude123412 commented:
I have just added permission5 to the pr, we are all ok to use some of there code here for this pr
![image](https://github.com/user-attachments/assets/8874ba48-7363-48ec-aae9-6e2262eeb951)

jude123412 committed:
Remove CosmicItemRenderer Class

Removed CosmicItemRenderer Class.
This has been Replaced By CosmicHaloItemRender from Avaritia.

Update Avaritia Dependencies

jude123412 commented:
I have Removed the CosmicItemRenderer Class and replaced it inside my own for of Avaritia

jude123412 committed:
Remove Avaritia Textures Import

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
Update Avaritia Dependancy

Remove Unnecessary option to set maskOpacity. Setting this Causes render bugs with the Infinity Armor Model.

Removed Fallback TextureAtlasSprite. Instead it is now set to null. when an Item isn't called with cosmic properties.

jude123412 commented:
I have updated some of the classes and fixed some bugs related with the Cosmic Render.
These bugs were brought to light following the fix of a similar bug In Avaritia :D

jude123412 committed:
Small tweaks

Removed the need to convert String to Integer when setting Halo Colour for MetaItems and Material Items.

jude123412 committed:
Description stuff and extra method I missed

Update description of how to use these new methods, explaining how to use the new hex values.

Added a method I missed for CosmicProperty.

jude123412 committed:
Forgot to Uncomment a line in dependencies.gradle

jude123412 committed:
Run SpotlessCheck/SpotlessApply

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
dependencies description

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
Update Mods.java

Fixed Mod check to only allow Avaritia methods when Avaritia 1.1x Unoffical Extended Life version 3.3.0.42 or greater is installed.
If you use the Covers Avaritia 1.1x these methods won't work and now the wont crash the game.

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10

jude123412 committed:
Merge branch 'master' into GTHH-GregTech-2.8.10
